player/lua.c

```diff
@@ -1194,17 +1167,64 @@ static const struct fn_entry utils_fns[] = {
     FN_ENTRY(split_path),
     FN_ENTRY(join_path),
     FN_ENTRY(getpid),
-    FN_ENTRY(parse_json),
-    FN_ENTRY(format_json),
+    AF_ENTRY(parse_json),
+    AF_ENTRY(format_json),
     {0}
 };
 
+/* returns the talloc ctx which script_autofree_trampoline sets */
+static void *mp_lua_PITA(lua_State *L)
+{
+    void **ctx = (void**)lua_touserdata(L, lua_upvalueindex(1));
+    if (!ctx) {
+        lua_pushstring(L, "missing autofree ctx");
+        lua_error(L);
+    }
+    return *ctx;
+}
+
+static int script_autofree_trampoline(lua_State *L)
+{
+    lua_CFunction target = lua_touserdata(L, lua_upvalueindex(1));
+    if (!target) {
+        lua_pushstring(L, "missing autofree target");
+        lua_error(L);
+    }
+
+    int nargs = lua_gettop(L);
+    void *ctx;
```

Comment:
No reason to make this separate.

Reply:
I don't understand. Separate from what?

Reply:
As discussed on irc, the init comes later because we first need the address of ctx, then do things which can cause lua errors, and only once we're done we're creating the talloc ctx. I.e. the init has to be separate.